plugins/org.eclipse.reddeer.jface/src/org/eclipse/reddeer/jface/matcher/WindowMatcher.java

```diff
@@ -12,46 +12,43 @@
 
 import org.eclipse.jface.window.Window;
 import org.eclipse.swt.widgets.Shell;
-import org.hamcrest.BaseMatcher;
 import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
 import org.eclipse.reddeer.common.exception.RedDeerException;
 import org.eclipse.reddeer.core.handler.WidgetHandler;
 /**
  * Matches shell which is JFace Window
  * @author rawagner
  *
- * @param <T> class that extends window
  */
-public class WindowMatcher<T extends Window> extends BaseMatcher<T>{
+public class WindowMatcher extends TypeSafeMatcher<Shell>{
 	
-	private Class<T> windowClass;
+	private Class<? extends Window> windowClass;
 	
-	public WindowMatcher(Class<T> windowClass) {
+	public <T extends Window> WindowMatcher(Class<T> windowClass) {
 		this.windowClass = windowClass;
 	}
 
 	@Override
-	public boolean matches(Object item) {
-		if(item instanceof Shell){
-			try{
-				Object data = WidgetHandler.getInstance().getData((Shell)item);
-				if(data != null) {
-					return windowClass.isInstance(data);
-				}
-			} catch (RedDeerException e) {
-				if(((Shell) item).isDisposed()){
-					return false;
-				}
-				throw e;
-			}
-		}
-		return false;
+	public void describeTo(Description description) {
+		description.appendText("Window matching class "+windowClass);
+		
 	}
 
 	@Override
-	public void describeTo(Description description) {
-		// TODO Auto-generated method stub
-		
+	protected boolean matchesSafely(Shell item) {
+		try{
+			Object data = WidgetHandler.getInstance().getData((Shell)item);
```

Comment:
bcs. item is already Shell, it is not necessary to cast Shell on it again. But that's just cosmetics :)